grammar.js

```diff
@@ -858,66 +854,67 @@ module.exports = grammar(C, {
     compound_literal_expression: ($, original) => choice(
       original,
       seq(
-        field('type', choice(
-          $._type_identifier,
-          $.template_type,
-          $.scoped_type_identifier
-        )),
+        field('type', $._class_name),
         field('value', $.initializer_list)
       )
     ),
 
-    scoped_field_identifier: $ => prec(1, seq(
+    dependent_identifier: $ => seq('template', $.template_function),
+    dependent_field_identifier: $ => seq('template', $.template_method),
+    dependent_type_identifier: $ => seq('template', $.template_type),
+
+    _scoped_namespace: $=> prec(1, seq(
       field('namespace', optional(choice(
         $._namespace_identifier,
         $.template_type,
-        $.scoped_namespace_identifier
+        alias($.dependent_type_identifier, $.dependent_name)
       ))),
       '::',
+    )),
+
+    scoped_field_identifier: $ => seq(
+      $._scoped_namespace,
       field('name', choice(
-        $._field_identifier,
-        $.operator_name,
-        $.destructor_name
+        alias($.dependent_field_identifier, $.dependent_name),
+        $.scoped_field_identifier,
```

Comment:
I am also considering aliasing the various "scoped_identifier" type rules so they are all simply "scoped_identifier" instead of having the 4 different rule names depending on the context and type of the inner-most rhs identifier. Thoughts?

Reply:
I think it’s a good idea.